server.js

```diff
@@ -21,9 +16,100 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
 });
 
+// Simple get request, sending back the complete list
+app.get("/streams", (req, res) => {
+  res.status(200).json({
+    data: streams,
+    success: true})
+})
+
+// Find one specific movie/show by name
+app.get('/streams/name/:title', (req, res) => {
+  const { title } = req.params
+
+  const netflixTitle = streams.find(
+  (stream) => stream.title.toLowerCase() === title.toLowerCase()
+  )
+
+  if(!netflixTitle){
+    res.status(404).json({
+      data: "This title is not found",
+      success: false })
+  } else {
+    res.status(200).json({
+      data: netflixTitle,
+      success: true })
+  }
+  })
+
+// Filter movies/shows by release year
+  app.get('/streams/year/:release_year', (req, res) => {
+    const { release_year } = req.params
+
+    const releaseYear = streams.filter(
+      (stream) => stream.release_year === +release_year
+    )
+
+  res.status(200).json({
+      data: releaseYear,
+      success: true })
+    
+  })
+
+
+// Filter by dynamic type, i.e. Movie or TV Show 
+app.get('/streams/type/:type', (req, res) => {
+  const { type } = req.params
+
+  const netflixTypes = streams.filter(
+  (stream) => stream.type.toLowerCase() === type.toLowerCase()
+  )
+    res.status(200).json({
+      data: netflixTypes,
+      success: true }) 
+  })
+
+// Filter by actors in movies and TV shows
+  app.get('/streams/actor/:cast', (req, res) => {
```

Comment:
I just wonder if it would be possible to filter multiple actors' names, just being curious!